bootstrap.c

```diff
@@ -81,25 +82,86 @@ static const char *get_host_triple(void) {
     return global_buffer;
 }
 
+#if defined(__WIN32__)
+#define HOST_PATHSEP '\\'
+#else
+#define HOST_PATHSEP '/'
+#endif
+
+static const char *get_parent_dir(const char *cwd) {
+    // safe for our use case
+    static char global_buffer[255];
+    if (strlen(cwd) + 3 >= 255)
+        panic("unable to get parent directory");
+
+    sprintf(global_buffer, "%s%c..", cwd, HOST_PATHSEP);
+    return global_buffer;
+}
+
+static const char *join(const char *dirpath, const char *path) {
+    int size = strlen(dirpath) + strlen(path) + 1;
+    if (size >= 4096)
+        panic("unable to join dirpath and path");
+
+    char *buffer = malloc(size);
+    sprintf(buffer, "%s%c%s", dirpath, HOST_PATHSEP, path);
+
+    // memory is leaked
```

Comment:
Would be nice, if valgrind would just work instead of having leaks cluttering output.

Reply:
I don't think it is a problem, since the memory will be reclaimed by the kernel.  I added the comment to make it clear that leaking is ok, but maybe it should be improved?

Thanks.

Reply:
Why should we use valgrind?